### PR TITLE
Fix some basic bug in AI

### PR DESCRIPTION
**Pull request by dnqbob**
I think those bugs should be very obvious so I put them together in a single PR.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs [hunk 1]
```diff
@@ -106,19 +106,12 @@ protected virtual bool ShouldFlee(Squad squad, Func<IReadOnlyCollection<Actor>,
...
-			if (activity.GetType() == typeof(Resupply))
-				return true;
-
-			var next = activity.NextActivity;
-			if (next == null)
-				return false;
-
-			if (next.GetType() == typeof(Resupply))
-				return true;
+			for (; activity != null; activity = activity.ChildActivity)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs]: Use
```csharp
return a.CurrentActivity.ActivitiesImplementing<Resupply>().Any();
```

and revert the change to Activity.cs (`ChildActivity` is deliberately kept non-public to avoid mod code from making a mess with its internals).

Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs]: looks like we also need a null check here.

Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs]: fixed

File: OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs [hunk 2]
```diff
@@ -106,21 +106,7 @@ protected virtual bool ShouldFlee(Squad squad, Func<IReadOnlyCollection<Actor>,
...
-
-			var next = activity.NextActivity;
-			if (next == null)
-				return false;
-
-			if (next.GetType() == typeof(Resupply))
-				return true;
-
-			return false;
+			return !a.IsIdle && a.CurrentActivity.ActivitiesImplementing<Resupply>().Any();
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs]: it should be checking for ReturnToBase activity as well, though both of these activities will be replaced by the docking refactor

File: OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs [hunk 3]
```diff
@@ -46,6 +47,9 @@ public class SquadManagerBotModuleInfo : ConditionalTraitInfo
 		[Desc("Own actor types that are prioritized when defending.")]
 		public readonly HashSet<string> ProtectionTypes = new();
 
+		[Desc("Target types are used for identifying aircraft.")]
+		public readonly BitSet<TargetableType> AircraftTargetType = new("Air", "AirborneActor");
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: this should have just `Air` type. The `AirborneActor` overwrite should only be applied to RA

Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs]: Fixed

File: OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs (hunk 2 above)
Comment (dnqbob) [reply OpenRA.Mods.Common/Traits/BotModules/Squads/States/StateBase.cs]: Fixed

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f5a5a0d66d58421507185a6a2527c86c69156465...a59d965e5a1f6312518c84e2338a243976c1853e)
